internal/kargo/kargo.go

```diff
@@ -114,3 +114,45 @@ func (p PromoWentTerminal) Update(e event.UpdateEvent) bool {
 	}
 	return false
 }
+
+// IgnoreClearRefreshUpdates is a predicate that filters out update events if
+// the update was only to clear the refresh annotation. This prevents the
+// extra reconciliation that happens when reconcilers respond to refresh by
+// updating the object's annotations.
+type IgnoreClearRefreshUpdates struct {
+	predicate.Funcs
+}
```

Comment:
This predicate eliminates the double reconcile that happens when user clicks refresh (which sets the refresh annotation), and then when the controller clears the refresh (which also modifies the annotation)

Reply:
Is there any chance of making this piece a separate PR? I'd rubber stamp that right away since I'd love having that improvement as soon as possible.

Reply:
Yea no problem. This PR may have overstepped a bit. I have a habit of fixing problems when it impacts the current work I'm doing.